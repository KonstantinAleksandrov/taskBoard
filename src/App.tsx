import { Table, ColumnCreator } from './containers';
import { TableContext } from './contexts';
import { tableStore } from './stores';

const App = () => {
  return (
    <TableContext.Provider value={tableStore}>
      <div className="App">
        <div className='wrapper'>
          <Table/>
          <ColumnCreator/>
        </div>
      </div>
    </TableContext.Provider>
  );
}

export default App;
