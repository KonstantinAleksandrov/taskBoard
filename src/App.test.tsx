import { render, screen ,fireEvent} from "@testing-library/react";
import App from './App';

describe('test App',()=>{
    test('наличие и работа кнопки по открыванию формы', ()=>{
        render(<App />);
        expect(screen.queryByTestId('add-form')).toBeNull()
        expect(screen.getByTestId('openFormButton')).toBeInTheDocument()
        
        fireEvent.click(screen.getByTestId('openFormButton'))
        expect(screen.getByTestId('add-form')).toBeInTheDocument()
        expect(screen.queryByTestId('openFormButton')).toBeNull()
    })

    test('клик на крестик в форме должен закрывать форму',()=> {
        render(<App />);
        fireEvent.click(screen.getByTestId('openFormButton'))
        const CloseCrossButton = screen.getByTestId('close-cross')
        expect(CloseCrossButton).toBeInTheDocument()

        fireEvent.click(CloseCrossButton)
        expect(screen.queryByTestId('add-form')).toBeNull()
        expect(screen.getByTestId('openFormButton') ).toBeInTheDocument()
    })

    test('проверка работы формы по добавлению колонки', ()=>{
        render(<App />);
        fireEvent.click(screen.getByTestId('openFormButton'))

        const addInput = screen.getByTestId('add-texarea')
        const addButton = screen.getByTestId('add-button')
        expect(addInput).toBeInTheDocument()
        expect(addButton).toBeInTheDocument()

        fireEvent.input(addInput, {
            target: {value: 'to do'}
        })
        fireEvent.click(addButton)
        const column = screen.getByTestId('column')
        const columnTitle = screen.getByTestId('column-title')
        expect(column).toBeInTheDocument()
        expect(columnTitle).toBeInTheDocument()
        expect(columnTitle).toHaveTextContent('to do')
        expect(screen.queryByTestId('add-form')).toBeNull()
    })

})
