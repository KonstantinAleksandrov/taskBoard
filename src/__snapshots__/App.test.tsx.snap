// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверяем наличие кнопки для добавления новой колонки на странице 1`] = `
<div
  class="add"
  title="addButton"
>
  <div
    class="addTitle"
  >
    + Add new column
  </div>
</div>
`;
